rs/nns/governance/src/storage/neurons.rs

```diff
@@ -819,31 +872,47 @@ fn update_repeated_field<Element, Memory>(
     update_range(new_entries, range, map)
 }
 
+fn insert_element_in_repeated_field<Element, Memory>(
+    neuron_id: NeuronId,
+    index: u64,
+    element: Element,
+    map: &mut StableBTreeMap<(NeuronId, /* index */ u64), Element, Memory>,
+) where
+    Element: Storable + PartialEq,
+    Memory: ic_stable_structures::Memory,
+{
+    let key = (neuron_id, index);
+    map.insert(key, element);
+}
+
 /// Replaces the contents of map where keys are in range with new_entries.
 // TODO(NNS1-2513): To avoid the caller passing an incorrect range (e.g. too
 // small, or to big), derive range from NeuronId.
 fn update_range<Key, Value, Memory>(
-    new_entries: HeapBTreeMap<Key, Value>,
+    mut new_entries: HeapBTreeMap<Key, Value>,
     range: impl RangeBounds<Key>,
     map: &mut StableBTreeMap<Key, Value, Memory>,
 ) where
     Key: Storable + Ord + Clone,
-    Value: Storable,
+    Value: Storable + PartialEq,
     Memory: ic_stable_structures::Memory,
 {
     let new_keys = new_entries.keys().cloned().collect::<HeapBTreeSet<Key>>();
 
+    let mut to_remove = vec![];
+    for (key, value) in map.range(range) {
+        if !new_keys.contains(&key) {
+            to_remove.push(key.clone());
+        } else if new_entries.get(&key).expect("We just checked it's there") == &value {
```

Comment:
It would be cleaner to branch on the value from the `get` than to use an `expect`.